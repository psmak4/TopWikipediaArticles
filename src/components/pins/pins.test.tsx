import { render, screen } from '@testing-library/react'
import Article from '~/models/article'
import ArticlesStore from '~/stores/articlesStore'
import Pins from './pins'

jest.mock('~/stores/articlesStore')
const mockArticlesStore = ArticlesStore as jest.MockedFunction<typeof ArticlesStore>
const mockCurrentPage = 1
const mockPageSize = 100
let mockPins = new Map<string, Article>()

describe('Pins', () => {
	beforeEach(() => {
		mockArticlesStore.mockImplementation(() => ({
			articles: [],
			currentPage: mockCurrentPage,
			pageSize: mockPageSize,
			pins: mockPins,
			searchDate: new Date(new Date().setDate(new Date().getDate() - 1)),
		}))
	})

	afterEach(() => {
		jest.clearAllMocks()
	})

	test('renders nothing when there are no pins', () => {
		render(<Pins />)

		const pinsElement = screen.queryByTestId('pins')
		expect(pinsElement).not.toBeInTheDocument()
	})

	test('renders pins when there are pins', () => {
		// Mock pins with some data
		mockPins = new Map([
			['key1', { article: 'Article 1', views: 100, rank: 1 }],
			['key2', { article: 'Article 2', views: 200, rank: 2 }],
		])

		render(<Pins />)

		const pinsElement = screen.queryByTestId('pins')
		expect(pinsElement).toBeInTheDocument()

		const articles = screen.getAllByRole('article')
		expect(articles.length).toBe(mockPins.size)
	})

	test('matches snapshot', () => {
		const { asFragment } = render(<Pins />)
		expect(asFragment()).toMatchSnapshot()
	})
})
