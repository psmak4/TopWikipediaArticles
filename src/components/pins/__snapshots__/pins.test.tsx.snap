// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pins matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pins"
    data-testid="pins"
  >
    <article
      class="article"
    >
      <h2
        class="title"
      >
        Article 1
      </h2>
      <div
        class="views"
      >
        100 views
      </div>
      <button
        class="pin"
      >
        <img
          alt="Unpin this article"
          src=""
        />
      </button>
    </article>
    <article
      class="article"
    >
      <h2
        class="title"
      >
        Article 2
      </h2>
      <div
        class="views"
      >
        200 views
      </div>
      <button
        class="pin"
      >
        <img
          alt="Unpin this article"
          src=""
        />
      </button>
    </article>
  </div>
</DocumentFragment>
`;
